asv/environment.py

```diff
@@ -77,6 +77,42 @@ def get_environments(conf):
             yield env
 
 
+def get_environment_class(conf, python):
+    """
+    Get a matching environment type class.
+
+    Parameters
+    ----------
+    conf : dict
+        asv configuration object
+
+    python : str
+        Python version specifier.  Acceptable values depend on the
+        Environment plugins installed but generally are:
+
+        - 'X.Y': A Python version, in which case conda or virtualenv
+          will be used to create a new environment.
+
+        - 'python' or '/usr/bin/python': Search for the given
+          executable on the search PATH, and use that.  It is assumed
+          that all dependencies and the benchmarked project itself are
+          already installed.
+    """
+    if conf.environment_type:
+        for cls in util.iter_subclasses(Environment):
+            if cls.tool_name == conf.environment_type:
+                return cls
+        raise ValueError(
+            "Unknown environment type '{0}'".format(conf.environment_type))
+    else:
+        # Try the subclasses in reverse order so custom plugins come first
+        for cls in list(util.iter_subclasses(Environment))[::-1]:
```

Comment:
Yes -- we should probably transition to being explicit.  I think in the meantime, I'll just add a warning.